### PR TITLE
Added support for seed in generating key pair

### PR DESCRIPTION
**Pull request by excerebrose**
This was done to unify the behaviour with the js-driver functionality

### CODE REVIEW

**Comment by ttmc** [comment]
Hi @excerebrose 

Before we can merge this pull request, we need you or your organization to agree to one of our contributor agreements. One of the big concerns for people using and developing open source software is that someone who contributed to the code might claim the code infringes on their copyright or patent. To guard against this, we ask all our contributors to sign a Contributor License Agreement. This gives us the right to use the code contributed and any patents the contribution relies on. It also gives us and our users comfort that they won't be sued for using open source software. We know it's a hassle, but it makes the project more reliable in the long run. Thank you for your understanding and your contribution!

If you are contributing on behalf of yourself (and not on behalf of your employer or another organization you are part of) then you should:

1. Go to: https://www.bigchaindb.com/cla/
2. Read the Individual Contributor Agreement
3. Fill in the form "For Individuals"
4. Check the box to agree
5. Click the SEND button

If you're contributing as an employee, and/or you want all employees of your employing organization to be covered by our contributor agreement, then someone in your organization with the authority to enter agreements on behalf of all employees must do the following:

1. Go to: https://www.bigchaindb.com/cla/
2. Read the Entity Contributor Agreement
3. Fill in the form "For Organizations”
4. Check the box to agree
5. Click the SEND button

We will email you (or your employer) with further instructions.

**Comment by ttmc** [comment]
@excerebrose I got an email saying that you agreed to the CLA and then I emailed you a block of text to paste into a comment here (as evidence that you're the same person). Once that's done, we can get this merged and do all the other things I indicated in https://github.com/bigchaindb/bigchaindb-driver/pull/487#issuecomment-431460996

**Comment by excerebrose** [comment]
This is to confirm that I agreed to and accepted the BigchainDB Individual Contributor Agreement at https://www.bigchaindb.com/cla/ and to represent and warrant that I have authority to do so.

XwyLR1hKbkDBU1Yuj2nreOJfA3FpMn9B5nk7pKtU5mhBnBpsJQQi7pHO0EUoW4p

File: cryptoconditions/crypto.py [hunk 1]
```diff
@@ -143,14 +147,19 @@ def encode(self, encoding='base58'):
         return super().encode(encoder=_get_nacl_encoder(encoding))
 
 
-def ed25519_generate_key_pair():
+def ed25519_generate_key_pair(seed=None):
     """
     Generate a new key pair.
 
+    Args:
+        seed (str) : seed for deterministic generation.
```
Comment (ttmc) [inline cryptoconditions/crypto.py]: I tried it and the `seed` must be a 32-byte `bytes` object (not a `str` object). For example:

```ipython
In [10]: myseed = open("/dev/urandom","rb").read(32)                                                                                                                                                        

In [11]: type(myseed)                                                                                                                                                                                       
Out[11]: bytes

In [12]: myseed                                                                                                                                                                                             
Out[12]: b'\x1b\xab\x1d\x98\x1d\xbdSl\xdbC\xf6\x1c\x18\xe7xYE=j=\xa9\xdd_\xcf\x0f\xad\xcf\xb2\x99=s\xca'

In [13]: keypair2 = crypto.ed25519_generate_key_pair(seed=myseed)                                                                                                                                           

In [14]: keypair2                                                                                                                                                                                           
Out[14]: 
(b'2s1Lv1saPMHbTL3F2m6dEfJ99oPtdEqd1bKCHDMvYJpR',
 b'9u98f5YaZ7eB6cVTQXjPgrNWeFF2Eqipb2VFZAo9Zrtq')
```

Comment (ttmc) [reply cryptoconditions/crypto.py]: That is, please update this line in the docstring to something like:
```text
        seed (bytes): 32-byte seed for deterministic generation.
```

**Comment by excerebrose** [comment]
Yup sorry, that's been fixed

**Comment by ttmc** [comment]
The Codecov tests are failing because this pull request added some new functionality (one new optional argument when generating a key pair) but didn't add any unit tests for that functionality. I'm not going to worry about that because this pull request was from an external contributor and I'd like to get their contribution merged ASAP. We can add the unit test in another PR.

**Comment by ttmc** [comment]
@excerebrose I'm going to merge this pull request now, and then I'll see if I can release a new version of the `cryptoconditions` package. I probably can. Assuming that works, I'll then turn my attention to https://github.com/bigchaindb/bigchaindb-driver/pull/487

**Pull request merged**